### PR TITLE
Allow patching of preview releases

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Allow patching of preview releases
  * release signed exe wrapper to storage
  * include prerelease iteration in file version to use in Microsoft Update detection.
  * Update tests to only generate pre-release iteration within the allowed range.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: PowerShell.Common.props [hunk 1]
```diff
@@ -74,6 +107,9 @@
         Lines="ReleaseTag=$(ReleaseTag);
                PowerShellVersion=$(PowerShellVersion);
                PSCoreBuildVersion = $(PSCoreBuildVersion);
+               PSCoreFileVersion = $(PSCoreFileVersion);
+               ReleaseTagVersionPart=$(ReleaseTagVersionPart);
+               ReleaseTagSemVersionPart=$(ReleaseTagSemVersionPart);
```
Comment (rjmholt) [inline PowerShell.Common.props:112]: We should be consistent about the spacing with `=` here

File: tools/ci.psm1 [hunk 2]
```diff
@@ -462,7 +462,8 @@ function Invoke-CIFinish
                 $previewLabel= "daily{0}" -f $previewLabel
             }
 
-            $preReleaseVersion = "$previewPrefix-$previewLabel.$env:BUILD_BUILDID"
+            $prereleaseIteration = (Get-Random -Minimum 1 -Maximum 99)
```
Comment (rjmholt) [inline tools/ci.psm1:465]: Perhaps rather than getting a random number, we could either peg this to `99` or use the last two digits of the `BUILD_ID`?

Comment (rjmholt) [reply tools/ci.psm1:465]: Alternatively, we could put the iteration here in the 200+ range by taking the last two digits and prepending a 2? That's more work, but would definitively mark an asset as CI-generated

Comment (rjmholt) [reply tools/ci.psm1:465]: My main concern is that adding randomness to builds isn't a good thing in terms of reproducibility

Comment (TravisEz13) [reply tools/ci.psm1:465]: From our perspective the build_ID is random.  

taking the last two digits would give things like 01, I'd have to cast that to an int, but that's possible.

The whole point here is to find these failures.  I don't think we should drop this.  If it becomes a problem, we can add a variable to override to be able to reproduce issues.

Comment (rjmholt) [reply tools/ci.psm1:465]: > The whole point here is to find these failures

Which failures are you referring to?

> I don't think we should drop this

I didn't follow this part. Drop the randomisation? In my mind we're dropping information by assigning a random number. What's the scenario you have in mind here that this `Get-Random` call serves?

> If it becomes a problem, we can add a variable to override to be able to reproduce issues

I don't foresee it becoming a problem, but I think we should have good justification for introducing a random number vs something deterministic.

Comment (TravisEz13) [reply tools/ci.psm1]: > What's the scenario you have in mind here that this Get-Random call serves? 
This replicates our incrementing the pre-release iteration during preview.

**Review by rjmholt — APPROVED** [review]

**Pull request merged**